scripts/DCOSWindowsAgentSetup.ps1

```diff
@@ -365,14 +330,58 @@ function New-DCOSServiceWrapper {
     if(!(Test-Path -Path $parent)) {
         New-Item -ItemType "Directory" -Path $parent
     }
-    Start-ExecuteWithRetry { Invoke-WebRequest -UseBasicParsing -Uri $SERVICE_WRAPPER_URL -OutFile $SERVICE_WRAPPER }
+    Write-Log "Copying $SERVICE_WRAPPER_FILE file"
+    Copy-item "$AGENT_BLOB_DEST_DIR\$SERVICE_WRAPPER_FILE" $SERVICE_WRAPPER -Force
     Write-Log "Exit New-DCOSServiceWrapper"
 }
 
+function Fetch-AgentBlobFiles {
+    Write-Log "Enter Fetch-AgentBlobFiles"
+    New-item $AGENT_BLOB_DEST_DIR -itemtype directory -ErrorAction SilentlyContinue
+
+    Write-Log "Download 7-Zip"
+    $7ZipFileName = "7z1801-x64.msi"
+    $7ZipMsiUrl = "$BootstrapUrl/$7ZipFileName"
+    $7ZipMsiFile = Join-Path $env:TEMP $7ZipFileName
+    Remove-item $7ZipMsiFile -ErrorAction SilentlyContinue
+    Write-Log "Downloading $7ZipMsiUrl to $7ZipMsiFile"
+    Measure-Command { curl.exe --keepalive-time 2 -fLsS --retry 10 -Y 100000 -y 60 -o $7ZipMsiFile $7ZipMsiUrl}
```

Comment:
Add a check for `$LASTEXITCODE` to ensure proper error checking for the `curl.exe` command.

Reply:
Will do